### PR TITLE
Update Kubernetes Azure Policy Addon Helm Charts

### PR DESCRIPTION
**Pull request by nreisch**
This PR updates the AKS Engine and Arc Helm Charts.

### CODE REVIEW

**Review by RamyasreeChakka — APPROVED** [review]
:shipit:

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/gatekeeper.yaml [hunk 1]
```diff
@@ -352,29 +764,32 @@ spec:
...
+            cpu: {{ .Values.gatekeeper.resources.webhookCPURequests }}
+            memory: {{ .Values.gatekeeper.resources.webhookMemoryRequests }}
         securityContext:
           allowPrivilegeEscalation: false
           capabilities:
             drop:
             - all
+          runAsGroup: 999
+          runAsNonRoot: true
+          runAsUser: 1000
```
Comment (RamyasreeChakka) [inline extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/gatekeeper.yaml:779]: Remove these as the values are not compatible with OpenShift.

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/gatekeeper.yaml [hunk 2]
```diff
@@ -284,39 +584,151 @@ spec:
...
+  replicas: {{ .Values.gatekeeper.resources.webhookReplicaCount }}
   selector:
     matchLabels:
       control-plane: controller-manager
+      gatekeeper.sh/operation: webhook
       gatekeeper.sh/system: "yes"
   template:
     metadata:
+      annotations:
+        container.seccomp.security.alpha.kubernetes.io/manager: runtime/default
```
Comment (RamyasreeChakka) [inline extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/gatekeeper.yaml:704]: Remove the annotation as it is not compatible with OpenShift.

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/gatekeeper.yaml [hunk 3]
```diff
@@ -284,39 +584,151 @@ spec:
...
+            cpu: {{ .Values.gatekeeper.resources.auditCPURequests }}
+            memory: {{ .Values.gatekeeper.resources.auditMemoryRequests }}
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - all
+          runAsGroup: 999
+          runAsNonRoot: true
+          runAsUser: 1000
```
Comment (RamyasreeChakka) [inline extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/gatekeeper.yaml:679]: Remove these as values not compatible with OpenShift

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/gatekeeper.yaml [hunk 4]
```diff
@@ -284,39 +584,151 @@ spec:
...
   replicas: 1
+  selector:
+    matchLabels:
+      control-plane: audit-controller
+      gatekeeper.sh/operation: audit
+      gatekeeper.sh/system: "yes"
+  template:
+    metadata:
+      annotations:
+        container.seccomp.security.alpha.kubernetes.io/manager: runtime/default
```
Comment (RamyasreeChakka) [inline extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/gatekeeper.yaml:609]: Remove annotation as its not compatible with OpenShift

**Review by RamyasreeChakka — APPROVED** [review]
LGTM

**Review by pilor — APPROVED** [review]

**Pull request merged**